litellm/llms/palm.py

```diff
@@ -118,17 +118,15 @@ def completion(
             k not in inference_params
         ):  # completion(top_k=3) > palm_config(top_k=3) <- allows for dynamic variables to be passed in
             inference_params[k] = v
-
     prompt = ""
+    # Construct messages with roles
+    palm_messages = []
+    context = inference_params.get("context", None)
     for message in messages:
-        if "role" in message:
-            if message["role"] == "user":
-                prompt += f"{message['content']}"
-            else:
-                prompt += f"{message['content']}"
-        else:
-            prompt += f"{message['content']}"
-
+        role = message.get("role", "user")
+        content = message["content"]
+        palm_messages.append({"author": role, "content": content})
+        prompt = {"context": context, "messages": palm_messages}
```

Comment:
@SujanShilakar check `prompt` it will only have last value from `messages` unlike before.

I think you should make prompt as list, then save all of them there.

Reply:
I have looked after it and resolved too.